File: 199-binary-tree-right-side-view/binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) { 
        vector<int> ans; 
        if(root!=NULL) Helper(root,0,ans);
        return ans;
    }
    void Helper(TreeNode* root, int level,vector<int>& ans){
        if(ans.size()==level) ans.push_back(root->val);
        if(root->right!=NULL) Helper(root->right,level+1,ans);
        if(root->left!=NULL) Helper(root->left,level+1,ans);
    }
};

File: 199-binary-tree-right-side-view/binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> ans = new ArrayList<>();
        if(root==null) return ans;
        Helper(root,ans,0);
        return ans;
    }
    public void Helper(TreeNode root,List<Integer> ans, int level){
        if(ans.size()==level){
            ans.add(root.val);
        }
        if(root.right!=null) Helper(root.right,ans,level+1);
        if(root.left!=null) Helper(root.left,ans,level+1);
    }
}